pkg/agent/route/route_linux.go

```diff
@@ -1551,6 +1551,26 @@ func (c *Client) deleteNodeIP(podCIDR *net.IPNet) error {
 	return nil
 }
 
+func (c *Client) AddRouteForLink(cidr *net.IPNet, linkIndex int) error {
+	route := &netlink.Route{
+		Scope:     netlink.SCOPE_LINK,
+		Dst:       cidr,
+		LinkIndex: linkIndex,
+	}
+
+	return c.netlink.RouteReplace(route)
+}
+
+func (c *Client) DeleteRouteForLink(cidr *net.IPNet, linkIndex int) error {
+	route := &netlink.Route{
+		Scope:     netlink.SCOPE_LINK,
+		Dst:       cidr,
+		LinkIndex: linkIndex,
+	}
+
+	return c.netlink.RouteDel(route)
```

Comment:
It should check the error is not NotFound, otherwise if the route doesn't exist somehow, the caller will keep retrying.

Reply:
A check is added, when got "no such process", which means the route does not exist, the function will return nil